code/API/FreshdeskAPI.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Psr7\Request;
+
+class FreshdeskAPI extends \Object
+{
+    /**
+     * This is used by the SilverStripe Injector to instantiate an object as $this->client. See _config/config.yml.
+     *
+     * @var array
+     */
+    private static $dependencies = [
+        'client' => '%$FreshdeskClient',
+    ];
+
+    /**
+     * Generic method to handle requests to the Freshdesk API.
+     *
+     * @param string $method, String $url, String $action, Array $headers, Array $data
+     * @param mixed  $url
+     * @param mixed  $action
+     * @param mixed  $data
+     * @param mixed  $headers
+     *
+     * @return $response || false
+     */
+    public function APICall($method, $url, $action, $data = [], $headers = ['Content-type' => 'application/json'])
+    {
+        $request = new Request($method, 'https://'.$url.$action, $headers);
+
+        try {
+            $response = $this->client->send($request, ['json' => $data]);
+        } catch (RequestException $e) {
+            if ($e->hasResponse()) {
+                SS_Log::log($e->getMessage(), SS_Log::ERR);
+            }
+            $response = false;
```

Comment:
We don't want to ignore the exception without surfacing any errors to the UI. It looks like in case of failure, for example in ticket listing, we will just get an empty list without any indication something went wrong. Let's surface the error to the user.

Reply:
ack